website/docs/language/checks/index.mdx

```diff
@@ -0,0 +1,83 @@
+
+The following example loads the Terraform website and validates it is returning the expected status code of 200.
+
+```hcl
+check "health_check" {
+  data "http" "terraform_io" {
+    url = "https://www.terraform.io"
+  }
+
+  assert {
+    condition = data.http.terraform_io.status_code == 200
+    error_message = "${data.http.terraform_io.url} returned an unhealthy status code"
+  }
+}
+```
+
+### Scoped Data Sources
+
+A `check` block can optionally have a nested data source. This block behaves in the same way as normal/external [data sources](/terraform/language/data-sources), except it cannot be referenced from outside the enclosing check block. In addition, any errors or failures raised by the provider of the data source will be masked as warnings within the output and will not prevent the continued execution of the overall Terraform operation.
+
+A scoped data source allows you to validate the status of a piece of infrastructure outside the usual Terraform resource lifecycle. In the example in the previous section, if the `terraform_io` data source fails to load we would simply receive a warning that the `http` block couldn't be loaded rather than the blocking error that would be raised if the data source was defined outside a check block.
+
+Any data source from any provider can be used within a check block.
+
+#### Meta-Arguments
+
+Scoped data sources support the `depends_on` and `provider` [meta-arguments](/terraform/language/resources/syntax#meta-arguments), but do not support the `count`, `foreach`, or `lifecycle` arguments.
+
+### Assertions
+
+Each `check` block must have at least one, but potentially many, `assert` blocks. An `assert` block has a `condition` attribute and an `error_message` attribute, that match the definitions within [Custom Conditions](/terraform/language/expressions/custom-conditions).
+
+Unlike other custom conditions, assertions do not affect the execution of the overall Terraform operation. A failed assertion will report a warning diagnostic without halting the ongoing operation. This is in contrast to other custom conditions, such as a resource postcondition, which will block future resources from being applied or planned.
+
+Conditions within assertions can refer to any scoped data source within the enclosing check block, and can refer to any variables, resources, data sources, or module outputs within the current module.
+
+Assertions are discussed more in detail within the [Custom Conditions](/terraform/language/expressions/custom-conditions#check-assertions) documentation.
+
+### Meta-Arguments
+
+In the initial release of checks, none of the traditional resource [meta-arguments](/terraform/language/resources/syntax#meta-arguments) are supported. We are interested to see how the community uses this new feature, so please file feedback with use-cases that would benefit from the introduction of any meta-arguments.
```

Comment:
It'd be nice to have a feedback link here, do folks generally file a github issues or something else? We could add a link?


```suggestion
Check blocks do not currently support [meta-arguments](/terraform/language/resources/syntax#meta-arguments). We are still collecting feedback on this feature, so if your use case would benefit from check blocks supporting meta-arguments, please let us know. 
```